heterocl/types.py

```diff
@@ -40,6 +40,9 @@ def __eq__(self, other):
         if other is None:
             return False
         other = dtype_to_hcl(other)
+        # check if self and other are the same type
+        if not isinstance(self, type(other)):
```

Comment:
Can you explicitly use `type(self) != type(other)`? Otherwise, they may have inherited relationship

Reply:
That makes sense, I changed to `if not type(self) is type(other)` since pylint complains about `!=`, `type` returns a class so I think `is` also works